src/main/java/com/google/devtools/build/lib/exec/SpawnStrategyRegistry.java

```diff
@@ -595,6 +631,51 @@ public String toString() {
     }
   }
 
+  private static class StrategyPlatformFilter {
+    private final StrategyMapper strategyMapper;
+    private final ImmutableListMultimap<Label, SpawnStrategy> platformToStrategies;
+
+    private StrategyPlatformFilter(
+      StrategyMapper strategyMapper,
+      ImmutableListMultimap<Label, SpawnStrategy> platformToStrategies) {
+      this.strategyMapper = strategyMapper;
+      this.platformToStrategies = platformToStrategies;
+    }
+
+    public <T extends SpawnStrategy> List<T> getStrategies(
```

Comment:
This logic is now complicated enough: add a comment explaining how the data from the different flags is combined (similar to the docs for the new flag), to help future readers understand the intent.

Reply:
Docs added.

While doing so I noticed a missed code branch in `getDynamicSpawnActionContexts` for `if (mnemonicToDynamicStrategies.containsKey("")) {` which is now properly handled.